samza-core/src/main/java/org/apache/samza/clustermanager/ContainerProcessManager.java

```diff
@@ -117,8 +131,28 @@ public ContainerProcessManager(Config config, SamzaApplicationState state, Metri
 
     ResourceManagerFactory factory = getContainerProcessManagerFactory(clusterManagerConfig);
     this.clusterResourceManager = checkNotNull(factory.getClusterResourceManager(this, state));
-    this.metrics = new ContainerProcessManagerMetrics(config, state, registry);
 
+    // Initialize metrics
+    this.containerProcessManagerMetrics = new ContainerProcessManagerMetrics(config, state, registry);
+    this.jvmMetrics = new JvmMetrics(registry);
+    this.metricsReporters = MetricsReporterLoader.getMetricsReporters(new MetricsConfig(config), METRICS_SOURCE_NAME);
+
+    // Creating diagnostics manager and reporter, and wiring it respectively
+    Optional<String> execEnvContainerId = Optional.ofNullable(System.getenv("CONTAINER_ID"));
```

Comment:
Is there a constant for this env property?

Reply:
Bump ^

Reply:
I defined one, but the principled way to fix this is to define an API in ClusterResourceManager called getContainerId() and have YarnClusterResourceManager (which is in samza-yarn) return it. 
I will do that change separately.